packages/nx-plugin/src/executors/internal/config.ts

```diff
@@ -0,0 +1,71 @@
+import { join } from 'node:path';
+import type { PersistConfig, UploadConfig } from '@code-pushup/models';
+import { slugify } from '../../internal/utils';
+import { parseEnv } from './env';
+import {
+  BaseNormalizedExecutorContext,
+  GlobalExecutorOptions,
+  ProjectExecutorOnlyOptions,
+} from './types';
+
+export function globalConfig(
+  options: Partial<GlobalExecutorOptions>,
+  context: BaseNormalizedExecutorContext,
+): Required<GlobalExecutorOptions> {
+  const { projectConfig } = context;
+  const { root: projectRoot = '' } = projectConfig ?? {};
+  // For better debugging use `--verbose --no-progress` as default
+  const { verbose, progress, config } = options;
+  return {
+    verbose: !!verbose,
+    progress: !!progress,
+    config: config ?? join(projectRoot, 'code-pushup.config.json'),
+  };
+}
+
+export function persistConfig(
+  options: Partial<PersistConfig & ProjectExecutorOnlyOptions>,
+  context: BaseNormalizedExecutorContext,
+): Partial<PersistConfig> {
+  const { projectConfig } = context;
+
+  const { name: projectName = '', root: projectRoot = '' } =
+    projectConfig ?? {};
+  const {
+    format = ['json'],
+    outputDir = join(projectRoot, '.code-pushup', projectName), // always in <root>/.code-pushup/<project-name>,
```

Comment:
I wanted to mirror a similar structure as with e.g. the coverage folder structure. (coverage/utils/unit-test)

<purpose>/<packages>/results/**

Reply:
For coverage we have `coverage/<project-name>/{unit,integration}-tests` - each folder there is needed to prevent overwriting content.

If I understand it correctly, this would save `report.{json,md}` in `packages/utils/.code-pushup/utils`, for example - the last `utils` folder seems unnecessary, because nothing else writes into `packages/utils/.code-pushup`.

But if we take `dist/...` and `coverage/...` output directories as inspiration, maybe it's a better option to save all project reports into a single location rather than have it in each project's folder. For example, each project would create `.code-pushup/<project-name>/report.{json,md}`, then it's easier to browse through many project reports just by opening the root `.code-pushup` folder. Also, having only one `.code-pushup` folder in the monorepo is a bit tidier :thinking:

Reply:
> packages/utils/.code-pushup/utils

Actually it should be in `.code-pushup/utils`. Will fix it!! Sry i did not see that one.